src/OpenTelemetry.Instrumentation.AWSLambda/SourceGenerationContext.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET6_0_OR_GREATER
+using System.Text.Json.Serialization;
+using Amazon.Lambda.SNSEvents;
+
+namespace OpenTelemetry.Instrumentation.AWSLambda;
+
+/// <summary>
+/// "Source Generation" is feature added to System.Text.Json in .NET 6.0.
+/// This is a performance optimization that avoids runtime reflection when performing serialization.
+/// Serialization metadata will be computed at compile-time and included in the assembly.
+/// <see href="https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/source-generation-modes" />.
+/// <see href="https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/source-generation" />.
+/// <see href="https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator/" />.
+/// </summary>
+[JsonSerializable(typeof(SNSEvent.SNSMessage))]
```

Comment:
The tests pass, but there's no `[JsonPropertyName]` attributes on this class, so I'm not 100% sure if the serialization is right or not.

Reply:
You generally only need to use `JsonPropertyName` on response objects where the casing on the outgoing JSON matters. When going from JSON to POCO we enable case insensitive matches in the serializer.